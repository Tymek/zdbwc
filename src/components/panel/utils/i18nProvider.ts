import polishMessages from '@scrlk/ra-language-polish'
import { TranslationMessages } from 'ra-core'
import polyglotI18nProvider from 'ra-i18n-polyglot'

const fields = {
	name: 'nazwa',
	subject: 'temat',
	begins_at: 'od',
	ends_at: 'do',
	speaker: 'mówca',
	session: 'sesja',
	description: 'opis',
	updated_at: 'zaktualizowano',
	created_at: 'utworzono',
	location: 'lokalizacja',
	title: 'tytuł',
	content: 'treść',
	published_at: 'data publikacji',
}

const messages = {
	pl: {
		...polishMessages,
		ra: {
			...polishMessages.ra,
			action: {
				...polishMessages.ra.action,
				create: 'Dodaj',
			},
			page: {
				...polishMessages.ra.page,
				dashboard: 'Statystyki',
			},
		},
		resources: {
			session: {
				name: 'Sesja |||| Sesje',
				fields,
			},
			topic: {
				name: 'Temat |||| Tematy',
				fields,
			},
			notification: {
				name: 'Powiadomienie |||| Powiadomienia',
				fields,
				empty: 'Brak powiadomień na liście',
				invite: 'Czy chcesz dodać nową wiadomość?',
			},
		},
	},
}

export default polyglotI18nProvider(locale => (locale === 'pl' ? messages[locale] : {} as TranslationMessages), 'pl')
