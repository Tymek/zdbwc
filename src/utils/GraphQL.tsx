import React, { FunctionComponent, useMemo } from 'react'
import isofetch from 'isomorphic-unfetch'
import {
	ApolloClient,
	ApolloLink,
	HttpLink,
	InMemoryCache,
	ApolloProvider,
	NormalizedCacheObject,
	QueryOptions,
} from '@apollo/client'

const ssrMode = typeof window === 'undefined'
export const pagePropsKey = 'initialApolloState'

let uri = ssrMode
	? 'http://hasura:8080/v1/graphql'
	: 'http://localhost:8080/v1/graphql'

if (process.env.NODE_ENV === 'development' && !ssrMode) {
	uri = `http://${window.location.hostname}:8080/v1/graphql`
}

if (process.env.NODE_ENV === 'test') {
	uri = 'http://localhost:3000/graphql'
}

let apolloClient: ApolloClient<NormalizedCacheObject>

export function gqlClient(initialState?: NormalizedCacheObject): ApolloClient<NormalizedCacheObject> {
	const client = apolloClient ?? new ApolloClient({
		ssrMode,
		link: ApolloLink.from([
			new HttpLink({
				uri,
				credentials: 'include', // NOTE: 'same-origin'?
				fetch: isofetch,
			}),
			// NOTE: error link?
		]),
		cache: new InMemoryCache(),
	})

	if (initialState) {
		client.cache.restore(initialState)
	}

	if (!ssrMode) {
		apolloClient = apolloClient ?? client
	}

	return client
}

type PageProps = {
	[pagePropsKey]?: NormalizedCacheObject
}

export const Provider:FunctionComponent<{pageProps: PageProps }> = ({ children, pageProps }) => {
	const initialState: NormalizedCacheObject | undefined = pageProps && pageProps[pagePropsKey]
	const client = useMemo(() => gqlClient(initialState), [initialState])

	return (
		<ApolloProvider client={client}>
			{children}
		</ApolloProvider>
	)
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const precacheQuery = (query: QueryOptions<Record<string, any>>) => async (): Promise<any> => {
	const client = gqlClient()

	await client.query(query)

	return {
		props: {
			[pagePropsKey]: client.cache.extract(),
		},
		unstable_revalidate: 1,
	}
}
