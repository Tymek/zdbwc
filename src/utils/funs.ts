const tst = (a: number): boolean => a === 1

export default tst
