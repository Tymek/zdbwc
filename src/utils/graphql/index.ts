export { default as gqlClient } from './client'
export { default as Provider } from './Provider'
export { default as precacheQueries } from './precacheQueries'
