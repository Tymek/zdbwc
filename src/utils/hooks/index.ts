/* eslint-disable import/prefer-default-export */
export { default as useMedia } from './useMedia'
