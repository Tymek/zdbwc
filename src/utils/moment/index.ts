import moment from 'moment-timezone'
import 'moment/locale/pl'

moment.locale('pl')
moment.tz.setDefault('Europe/Warsaw')

export default moment
