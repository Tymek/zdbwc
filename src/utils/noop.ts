// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-empty-function
export default () => {}
