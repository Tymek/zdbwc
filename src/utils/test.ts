import * as React from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import Providers from './providers'

const customRender = (
    ui: React.ReactElement,
    options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: Providers as React.ComponentType, ...options })

// eslint-disable-next-line import/export
export * from '@testing-library/react'
// eslint-disable-next-line import/export
export { customRender as render }
