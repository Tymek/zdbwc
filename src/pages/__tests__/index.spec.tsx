import * as React from 'react'
import { render, mocked, waitFor, screen } from 'utils/test'
// import Providers from 'utils/providers'
import Head from 'next/head'

import Index from '..'

jest.mock('next/head')
const mockedHead = mocked(Head)

describe('pages/index', () => {
	it('loads and displays greeting', async () => {
		render(<Index />)

		await waitFor(() => screen.getByRole('main'))

		expect(mockedHead).toHaveBeenCalledTimes(1)
		expect(mockedHead.mock.calls[0]).toMatchSnapshot()
		expect(screen.getByRole('heading')).toHaveTextContent('Welcome')
	})
})
