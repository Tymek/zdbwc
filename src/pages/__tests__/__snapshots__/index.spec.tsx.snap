// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/index loads and displays greeting 1`] = `
Array [
  Object {
    "children": Array [
      <title>
        App
      </title>,
      <link
        href="/favicon.ico"
        rel="icon"
      />,
    ],
  },
  Object {},
]
`;
